apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java

```diff
@@ -591,20 +637,32 @@ void testOutputDevices() throws InterruptedException {
                 int[] channelCounts = deviceInfo.getChannelCounts();
                 numTested++;
                 // Always test mono and stereo.
-                testOutputDeviceCombo(id, deviceType, 1, 0);
-                testOutputDeviceCombo(id, deviceType, 2, 0);
-                testOutputDeviceCombo(id, deviceType, 2, 1);
+                testOutputDeviceCombo(id, deviceType, 1, 0, false);
+                testOutputDeviceCombo(id, deviceType, 2, 0, false);
+                testOutputDeviceCombo(id, deviceType, 2, 1, false);
                 if (channelCounts.length > 0) {
                     for (int numChannels : channelCounts) {
                         // Test higher channel counts.
                         if (numChannels > 2) {
                             log("numChannels = " + numChannels + "\n");
                             for (int channel = 0; channel < numChannels; channel++) {
-                                testOutputDeviceCombo(id, deviceType, numChannels, channel);
+                                testOutputDeviceCombo(id, deviceType, numChannels, channel, false);
                             }
                         }
                     }
                 }
+                int[] channelMasks = deviceInfo.getChannelMasks();
+                if (channelMasks.length > 0) {
+                    for (int channelMask : channelMasks) {
+                        int nativeChannelMask = channelMask >> 2;
```

Comment:
shift?!

Reply:
Yes, this is really bizarre but intentional. I added a comment.